### PR TITLE
updating results processing and tests

### PR DESCRIPTION
**Pull request by michaelbynum**

### CODE REVIEW

File: pyomo/contrib/solver/solution.py [hunk 1]
```diff
@@ -142,29 +143,48 @@ def __init__(self, sol_data: SolFileData, nl_info: NLWriterInfo) -> None:
...
     ) -> NoReturn:
-        if self._nl_info.scaling:
-            for v, val, scale in zip(
-                self._nl_info.variables, self._sol_data.primals, self._nl_info.scaling
-            ):
-                v.set_value(val / scale, skip_validation=True)
+        if self._nl_info is None:
+            raise RuntimeError(
+                'Solution loader does not currently have a valid solution. Please '
+                'check the termination condition.'
```
Comment (mrmundt) [inline pyomo/contrib/solver/solution.py:149]: I'm going to change this phrasing a bit after merging, just FYI.

File: pyomo/contrib/solver/solution.py [hunk 2]
```diff
@@ -142,29 +143,48 @@ def __init__(self, sol_data: SolFileData, nl_info: NLWriterInfo) -> None:
...
 
     def get_primals(
         self, vars_to_load: Optional[Sequence[_GeneralVarData]] = None
     ) -> Mapping[_GeneralVarData, float]:
-        if self._nl_info.scaling is None:
-            scale_list = [1] * len(self._nl_info.variables)
-        else:
-            scale_list = self._nl_info.scaling.variables
+        if self._nl_info is None:
+            raise RuntimeError(
```
Comment (mrmundt) [inline pyomo/contrib/solver/solution.py:172]: I'm going to change this phrasing a bit after merging, just FYI.

File: pyomo/contrib/solver/solution.py [hunk 3]
```diff
@@ -184,18 +204,28 @@ def get_primals(
     def get_duals(
         self, cons_to_load: Optional[Sequence[_GeneralConstraintData]] = None
     ) -> Dict[_GeneralConstraintData, float]:
+        if self._nl_info is None:
+            raise RuntimeError(
+                'Solution loader does not currently have a valid solution. Please '
```
Comment (mrmundt) [inline pyomo/contrib/solver/solution.py:209]: I'm going to change this phrasing a bit after merging, just FYI.

File: pyomo/contrib/solver/solution.py [hunk 4]
```diff
@@ -184,18 +204,28 @@ def get_primals(
...
+                'check the termination condition.'
+            )
+        if len(self._nl_info.eliminated_vars) > 0:
+            raise NotImplementedError('For now, turn presolve off (opt.config.writer_config.linear_presolve=False) to get dual variable values.')
+        assert self._sol_data is not None
```
Comment (mrmundt) [inline pyomo/contrib/solver/solution.py]: Should this be in here?

Comment (michaelbynum) [reply pyomo/contrib/solver/solution.py]: Is there a reason not to?

Comment (mrmundt) [reply pyomo/contrib/solver/solution.py]: If the assertion is false and a user hits it, it's going to error without a helpful message. I don't have a problem with assertions, but I do have a problem with non-error-messages.

Comment (michaelbynum) [reply pyomo/contrib/solver/solution.py]: fair. How is this?

Comment (mrmundt) [reply pyomo/contrib/solver/solution.py]: Better. I may tweak it a bit, but looks good!

**Pull request merged**